libsolidity/ast/Types.cpp

```diff
@@ -296,6 +296,24 @@ string Type::identifier() const
 	return ret;
 }
 
+TypeResult Type::binaryOperatorResult(Token _operator, const Type *_other) const
```

Comment:
I think this should be inside: 
```cpp 
TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other) const
```

Reply:
It is my IDE forcing const to be in front when I auto generate function, so it just should be like this here
```suggestion
TypeResult Type::binaryOperatorResult(Token _operator, Type const* _other) const
```

Reply:
Still, it should be `IntegerType`, not `Type`.

Reply:
https://github.com/ethereum/solidity/blob/77796d0349b65c1487ec12b3418a0de6ed0ca253/libsolidity/ast/Types.cpp#L596